batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/dao/StudentDao.java

```diff
@@ -0,0 +1,203 @@
+      ps.setString(2, stud.getAddress());
+      ps.setDate(3, new java.sql.Date(stud.getDob().getTime()));
+      ps.setString(4, stud.getDepartment());
+      ps.setString(5, stud.getBatch());
+      ps.setInt(6, stud.getRoll());
+      ps.executeUpdate();
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    } finally {
+      DbUtils.closePreparedStatement(ps);
+      DbUtils.closeConnection(conn);
+    }
+
+  }
+
+  public void delete(Integer studentId) throws DataException {
+    Connection conn = null;
+    PreparedStatement ps = null;
+    try {
+      conn = DbUtils.getConnection();
+
+      String query = "DELETE FROM student WHERE id=?";
+      ps = conn.prepareStatement(query);
+      ps.setInt(1, studentId);
+      ps.executeUpdate();
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    } finally {
+      DbUtils.closePreparedStatement(ps);
+      DbUtils.closeConnection(conn);
+    }
+  }
+
+  public void update(Student student) throws DataException {
+    Connection conn = null;
+    PreparedStatement ps = null;
+    try {
+      conn = DbUtils.getConnection();
+
+      String studentQuery = "UPDATE student SET name=?, address=?, dob=?, department=?, batch=?, roll=? WHERE id=?";
+      ps = conn.prepareStatement(studentQuery);
+      ps.setString(1, student.getName());
+      ps.setString(2, student.getAddress());
+      ps.setDate(3, new java.sql.Date(student.getDob().getTime()));
+      ps.setString(4, student.getDepartment());
+      ps.setString(5, student.getBatch());
+      ps.setInt(6, student.getRoll());
+      ps.setInt(7, student.getId());
+      ps.executeUpdate();
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    } finally {
+      DbUtils.closePreparedStatement(ps);
+      DbUtils.closeConnection(conn);
+    }
+  }
+
+  public Student fetchById(Integer id) throws DataException {
+    Connection conn = null;
+    PreparedStatement ps = null;
+    ResultSet studentResult = null;
+    try {
+      String query = "SELECT * FROM student WHERE id=?";
+      conn = DbUtils.getConnection();
+      ps = conn.prepareStatement(query);
+      ps.setInt(1, id);
+
+      studentResult = ps.executeQuery();
+
+      if (studentResult.next()) {
+        Student student = new Student();
+        student.setId(studentResult.getInt("id"));
+        student.setRoll(studentResult.getInt("roll"));
+        student.setDepartment(studentResult.getString("department"));
+        student.setBatch(studentResult.getString("batch"));
+
+        student.setName(studentResult.getString("name"));
+        student.setAddress(studentResult.getString("address"));
+        student.setDob(studentResult.getDate("dob"));
+
+        return student;
+      } else {
+        return null;
```

Comment:
instead of returning null, define student outside of if stmt using null and initialize inside if and return student object at the end.